core/src/main/scala/kafka/log/LogLoader.scala

```diff
@@ -106,7 +174,17 @@ object LogLoader extends Logging {
       loadSegmentFiles(params)
     })
 
-    completeSwapOperations(swapFiles, params)
+    // Do the actual recovery for toRecoverSwapFiles, as discussed above.
```

Comment:
Hmm, I am not sure why we need this step. We have processed all .swap files before and no new .swap files should be introduced if we get to here.

Reply:
you are right if we don't need to do sanity checks. Removed this